src/map.js

```diff
@@ -649,7 +647,30 @@ var map = function (arg) {
         height: m_height
       };
     }
-    m_this.resize(0, 0, arg.width, arg.height);
+    // store the original center and restore it after the resize
+    var oldCenter = m_this.center();
```

Comment:
Unless there is some reason to call `size` without some of the side effects (emitting a `resize` event and marking the map as modified), I don't see any benefit to calling another function here.